packages/paste-core/components/timeline/src/TimelineItem.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box } from "@twilio-paste/box";
+import { css, styled } from "@twilio-paste/styling-library";
+import React from "react";
+
+import { TimelineContext, TimelineGroupContext } from "./TimelineContext";
+import { TimelineItemCollapsible } from "./TimelineItemCollapsible";
+import { TimelineItemIcon } from "./TimelineItemIcon";
+import type { TimelineItemProps } from "./types";
+
+const TimelineItem = React.forwardRef<HTMLDivElement, TimelineItemProps>(
+  (
+    { children, icon, timestamp, title, collapsible = false, collapsibleHeading, element = "TIMELINE_ITEM", ...props },
+    ref,
+  ) => {
+    const { orientation } = React.useContext(TimelineContext);
+    const isGrouped = React.useContext(TimelineGroupContext);
+
+    if (!orientation) {
+      throw new Error("TimelineItem must be used within a Timeline component");
+    }
+
+    const ContainerStyled = styled.li(
+      css({
+        "&:last-child>div:first-child::after": {
+          content: "none",
+        },
+        "div::after": {
+          display: isGrouped ? "none" : "block",
+        },
+      }),
+    );
+
+    return (
+      <Box
+        element={element}
+        // @ts-expect-error we don't have polymorphic box typings yet
+        as={ContainerStyled}
+        ref={ref}
+        position="relative"
+        display="flex"
+        columnGap="space50"
+        rowGap="space50"
+        flexShrink={0}
+        flexDirection={orientation === "horizontal" ? "column" : "row"}
+        paddingX="space0"
+        {...props}
+      >
+        {!isGrouped ? (
+          <Box
+            element={`${element}_SEPARATOR`}
+            display="flex"
+            width={orientation === "horizontal" ? "initial" : "sizeBase50"}
+            flexDirection={orientation === "horizontal" ? "row" : "column"}
+            alignItems="center"
+          >
+            <TimelineItemIcon icon={icon} />
```

Comment:
```suggestion
            <TimelineItemIcon icon={icon} element=`{element}_ICON` />
```
Since we're not exposing TimelineItemIcon in the API , let's pass element down through the TimelineItem